pkg/manager/member/pd_scaler.go

```diff
@@ -74,15 +74,15 @@ func (s *pdScaler) ScaleOut(meta metav1.Object, oldSet *apps.StatefulSet, newSet
 	}
 
 	healthCount := 0
-	totalCount := *oldSet.Spec.Replicas
+	totalCount := len(tc.Status.PD.Members)
 	podOrdinals := helper.GetPodOrdinals(*oldSet.Spec.Replicas, oldSet).List()
 	for _, i := range podOrdinals {
 		targetPdName := PdName(tcName, i, tc.Namespace, tc.Spec.ClusterDomain)
 		if member, ok := tc.Status.PD.Members[targetPdName]; ok && member.Health {
 			healthCount++
 		}
 	}
-	if healthCount < int(totalCount) {
+	if healthCount < totalCount {
```

Comment:
Suggest removing the check here.
Why do we need to check the cluster status during scale out?

Reply:
This is introduced in https://github.com/pingcap/tidb-operator/pull/142 before the discovery service is introduced in https://github.com/pingcap/tidb-operator/pull/262.

Reply:
What's the problem in scaling when there are some unhealthy members for PD?
I think we can keep the check here and make a survey on this situation.